src/main/kotlin/uk/gov/gdx/datashare/repository/EgressEventDataRepository.kt

```diff
@@ -21,20 +20,36 @@ interface EgressEventDataRepository : CoroutineCrudRepository<EgressEventData, U
     consumerSubscriptionId: UUID,
     fromTime: LocalDateTime,
     toTime: LocalDateTime,
-  ): Flow<EgressEventData>
+  ): List<EgressEventData>
 
   @Query(
     "SELECT ed.* FROM egress_event_data ed " +
       "WHERE ed.when_created > :fromTime " +
       "AND ed.when_created <= :toTime " +
       "AND ed.consumer_subscription_id IN (:consumerSubscriptionIds) " +
-      "ORDER BY ed.when_created",
+      "ORDER BY ed.when_created " +
+      "LIMIT :pageSize " +
+      "OFFSET :offset",
+  )
+  fun findPageByConsumerSubscriptions(
```

Comment:
If we're using CrudRepository, do we need the queries here, and can we use Pageable instead of pageSize/offset? This might be better as a future change though